code/datums/achievements/_awards.dm

```diff
@@ -66,21 +66,58 @@
 /datum/award/proc/on_unlock(mob/user)
 	return
 
+///returns additional ui data for the Check Achievements menu
+/datum/award/proc/get_ui_data()
+	return list(
+		"score" = FALSE,
+		"achieve_info" = null,
+		"achieve_tooltip" = null,
+	)
+
 ///Achievements are one-off awards for usually doing cool things.
 /datum/award/achievement
 	desc = "Achievement for epic people"
 	icon = "" // This should warn contributors that do not declare an icon when contributing new achievements.
+	///How many players have earned this achievement
+	var/times_achieved = 0
+
+/datum/award/achievement/New()
+	. = ..()
+	var/datum/db_query/query = SSdbcore.NewQuery(
```

Comment:
This shouldn't be done for each achievement in the new, we should be doing one query for all achievement data during a subsystem initialization and then caching the results

Reply:
Done as you said, though I'm not storing the values in the achievement metadata table or anything yet while it still hasn't been testmerged.